src/CommunityToolkit.Maui.MediaElement.Analyzers.CodeFixes/UseCommunityToolkitMediaElementInitializationAnalyzerCodeFixProvider.cs

```diff
@@ -0,0 +1,76 @@
+using System.Reflection.Metadata.Ecma335;
+using System.Runtime.InteropServices.ComTypes;
+using System.Threading;
+using System.Threading.Tasks;
+using CommunityToolkit.Maui.MediaElement.Analyzers;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Rename;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace CommunityToolkit.Maui.Analyzers;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(UseCommunityToolkitMediaElementInitializationAnalyzerCodeFixProvider)), Shared]
+public class UseCommunityToolkitMediaElementInitializationAnalyzerCodeFixProvider : CodeFixProvider
+{
+	public sealed override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(UseCommunityToolkitMediaElementInitializationAnalyzer.DiagnosticId);
+
+	public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+	public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+	{
+		var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+		var diagnostic = context.Diagnostics.First();
+		var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+		// Find the type declaration identified by the diagnostic.
+		var declaration = root?.FindToken(diagnosticSpan.Start).Parent?.AncestorsAndSelf().OfType<InvocationExpressionSyntax>().Last() ?? throw new InvalidOperationException();
+
+		// Register a code action that will invoke the fix.
+		context.RegisterCodeFix(
+			CodeAction.Create(
+				title: CodeFixResources.Initialize__NET_MAUI_Community_Toolkit_MediaElement_Before_UseMauiApp,
+				createChangedDocument: c => AddUseCommunityToolkit(context.Document, declaration, c),
+				equivalenceKey: nameof(CodeFixResources.Initialize__NET_MAUI_Community_Toolkit_MediaElement_Before_UseMauiApp)),
+			diagnostic);
+	}
+
+	static async Task<Document> AddUseCommunityToolkit(Document document, InvocationExpressionSyntax invocationExpression, CancellationToken cancellationToken)
+	{
+		var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+		if (root is null)
+		{
+			return document;
+		}
+
+		var updatedInvocationExpression =
+			InvocationExpression(
+				MemberAccessExpression(
+					SyntaxKind.SimpleMemberAccessExpression, invocationExpression, IdentifierName("UseMauiCommunityToolkitMediaElement")));
+
+		var mauiCommunityToolkitUsingStatement =
+			UsingDirective(
+				QualifiedName(
+					IdentifierName("CommunityToolkit"),
+					IdentifierName("Maui")));
+
+		var newRoot = root.ReplaceNode(invocationExpression, updatedInvocationExpression);
+
+		if (newRoot is CompilationUnitSyntax compilationSyntax)
+		{
+			newRoot = compilationSyntax.AddUsings(mauiCommunityToolkitUsingStatement).NormalizeWhitespace();
+		}
```

Comment:
- This will add unnecessary using if the `using` is already a global using in another tree.
- normalizing the whole tree isn't a good idea.

Best is to let Roslyn do that via `AddImportsAnnotation`.

Reply:
Great suggestions, thanks. This will then be the same for the regular CodeFix project. Let's open an issue, or even better, a PR, for these fixes once this is merged.